### PR TITLE
Clone the parent Android engine's asset provider into a spawned engine

### PR DESCRIPTION
**Pull request by jason-simmons**
Typically the AndroidShellHolder's apk_asset_provider_ is set by AndroidShellHolder::Launch when the DartExecutor runs the engine's entrypoint.

But if the engine was started by Spawn, then the apk_asset_provider_ was not being set.  This would cause a crash if the resulting engine was later used to spawn another engine.

Fixes https://github.com/flutter/flutter/issues/122364

### CODE REVIEW

File: shell/platform/android/android_shell_holder.cc
Comment (reidbaker) [inline shell/platform/android/android_shell_holder.cc:1]: Does this need a test?

**Review by reidbaker — APPROVED** [review]
Rubber stamp.

**Comment by Hixie** [comment]
Do we test the Spawn codepaths in general?

**Comment by jason-simmons** [comment]
Added a test for this in the Android scenario_app

**Review by reidbaker — APPROVED** [review]

**Pull request merged**